src/main/java/org/spongepowered/api/CatalogKey.java

```diff
@@ -119,14 +120,39 @@ static CatalogKey of(final PluginContainer container, final String value) {
     /**
      * Resolves a catalog key from a string.
      *
-     * <p>If no namespace is found in {@code string} then
+     * <p>If no namespace is found in {@code value} then
      * {@link #MINECRAFT_NAMESPACE} will be the namespace.</p>
      *
      * @param value The value
      * @return A new catalog key
      */
     static CatalogKey resolve(final String value) {
-        return Sponge.getRegistry().getBuilderRegistry().provideBuilder(Builder.class).value(value).build();
+        return Sponge.getRegistry().getFactoryRegistry().provideFactory(Factory.class).resolve(value, MINECRAFT_NAMESPACE);
```

Comment:
```suggestion
        return resolve(value, MINECRAFT_NAMESPACE);
```
Reuse your existing logic for fallback handling.